submittal/src/main/java/com/kyriba/submittal/service/TopicServiceImpl.java

```diff
@@ -0,0 +1,43 @@
+package com.kyriba.submittal.service;
+
+import com.kyriba.submittal.dao.TopicRepository;
+import com.kyriba.submittal.domain.Topic;
+import com.kyriba.submittal.service.exception.ObjectNotFoundException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+
+/**
+ * @author M-ABL
+ */
+@Service
+@Transactional
+@RequiredArgsConstructor
+public class TopicServiceImpl implements TopicService
+{
+  private final TopicRepository topicRepository;
+
+
+  @Override
+  public void approve(final long topicId)
+  {
+    Topic topic = findTopicById(topicId);
+    topic.approve();
+  }
+
+
+  @Override
+  public void reject(final long topicId)
+  {
+    Topic topic = findTopicById(topicId);
+    topic.reject();

```

Comment:
shouldn't you call topicRepository.save(topic)?

Reply:
just removed it :) haven't you read @xpinjection notes? :)